pkg/genericapiserver/genericapiserver.go

```diff
@@ -207,30 +206,48 @@ type Config struct {
 
 // GenericAPIServer contains state for a Kubernetes cluster api server.
 type GenericAPIServer struct {
-	// "Inputs", Copied from Config
+	// ServiceClusterIPRange is used to build cluster IPs for discovery.  It is exposed so that `master.go` can
```

Comment:
Reason why this is also in GenericApiServer and not just in kube-apiserver is because this is also used by generic apiserver to detect clients that are within the cluster and hence can reach the apiserver with internal IP.
While returning the server address in discovery information, we return a CIDR which says that if the client IP is within --service-cluster-ip-range, then it should use server's internal IP. Else the client needs to use external IP.
Link to code: https://github.com/kubernetes/kubernetes/blob/da53a247633cd91bd8e9818574279f3b04aed6a5/pkg/genericapiserver/genericapiserver.go#L806